### ISSUE TITLE
UnusedVariable false positive

### ISSUE THREAD

**Issue by BenMorel** [vimeo/psalm#5957]
Hi, for the following code (simplified example from [brick/math](https://github.com/brick/math/blob/master/src/BigInteger.php#L231)):

```php
function randomBits() : string
{
    $bitmask = \chr(0xFF >> 1);

    $randomBytes    = random_bytes(1);
    $randomBytes[0] = $randomBytes[0] & $bitmask;

    return $randomBytes;
}
```

Psalm complains:

> INFO: UnusedVariable - 4:5 - $bitmask is never referenced or the value is not used

https://psalm.dev/r/e436f35229

**Comment by weirdan** [comment]
Seems to be caused by all bit operations on strings. Bit inversion (`~`) causes this for ints as well, but only when result is assigned to a variable: https://psalm.dev/r/ac6ce2540a
